baybe/acquisition/acqfs.py

```diff
@@ -1,11 +1,106 @@
 """Available acquisition functions."""
 
+import math
 from typing import ClassVar
 
+import pandas as pd
+from attr.validators import optional as optional_v
 from attrs import define, field
-from attrs.validators import ge, instance_of
+from attrs.validators import ge, gt, instance_of, le
 
 from baybe.acquisition.base import AcquisitionFunction
+from baybe.searchspace import SearchSpace
+from baybe.utils.sampling_algorithms import (
+    SamplingMethod,
+    sample_numerical_df,
+)
+
+
+########################################################################################
+### Active Learning
+@define(frozen=True)
+class qNegIntegratedPosteriorVariance(AcquisitionFunction):
+    """Monte Carlo based negative integrated posterior variance.
+
+    This is particularly useful for active learning as it is a measure for global model
+    uncertainty.
+    """
+
+    abbreviation: ClassVar[str] = "qNIPV"
+
+    sampling_fraction: float = field(
+        converter=float,
+        validator=[gt(0.0), le(1.0)],
+        default=1.0,
+    )
+    """Fraction of data that will be sampled for integrating the posterior.
+
+    The fraction will be ignored if 'sampling_n_points' is not None."""
```

Comment:
but I also want to retain the case that neither of them is provided in which case it will simply assume frac=1.0 (prob the default usage case), is that possible?

Reply:
I pushed a solution via `__attrs_post_init__` but it involves an ugly `object.__setattr__`, otherwise Id ont know to to handle it in the frozen setting

also had to change the hypothesis due to that additional constraint

did also not go with the xor because mypy complains about the None possibility

Reply:
@AdrianSosic I tested this new acqf with the examples fur purely conti and hybrid spaces. Conti seems to work, for hybrid I get this strange warning: any idea?
<img width="1502" alt="image" src="https://github.com/emdgroup/baybe/assets/17951239/3e4f03ba-3cfc-4edf-a488-abd114129fba">

Reply:
Off the top of my head: unfortunately no. Seems some botorch diggin is needed again =/

Reply:
Regarding the validation issue: I guess you were thinking too complicated. Have fixed it in a new commit + also fixed the validators and added validation tests. Let me know if you are happy with that version

Reply:
hmm I could swear I tried that and got some error inside the default decorator that theres no access to `sampling_n_points` (depsite changing the order too), together with this I decided against it: 
<img width="830" alt="image" src="https://github.com/emdgroup/baybe/assets/17951239/c8239754-3130-4cb5-aad2-bbc2c90238cb">
but seems to work fine now, just dont see why

Reply:
Hm, can't say how your version was different. But I'm quite confident that the current one is OK. I'm aware of the stuff from the screenshot, but that's exactly why the order matters. It has the nice side-effect that it forces you to put dependent attributes at the end, which automatically gives you a more "reasonable" signature. So from my end, everything is fine and you could close .... but is there still the problem with the warning?

Reply:
Botorch bug is confirmed, hence closing this thread.